### PR TITLE
Use valid default for SMTP protocol

### PR DESCRIPTION
**Pull request by fkrone**
The setting value has to be in upper case.

Tags: smtp, configuration

### CODE REVIEW

File: src/main/resources/component-web.conf [hunk 1]
```diff
@@ -351,7 +351,7 @@ mail {
         port = "25"
 
         # Protocol and encryption of the SMTP connection (SMTP, SMTPS or STARTTLS)
-        protocol = "smtp"
+        protocol = "SMTP"
```
Comment (andyHa) [inline src/main/resources/component-web.conf]: can't we rather/also make the reader smarter?

Comment (D3nnisH) [reply src/main/resources/component-web.conf]: enum values are case insensitive, so this wouldn't be unambiguously anymore

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 2]
```diff
@@ -89,7 +91,14 @@ public SMTPConfiguration(String host,
     public static SMTPConfiguration fromConfig() {
         return new SMTPConfiguration(smtpHost,
                                      smtpPort,
-                                     smtpProtocol,
+                                     Sirius.getSettings()
+                                           .get("mail.smtp.protocol")
+                                           .getEnum(SMTPProtocol.class)
+                                           .orElseGet(() -> Value.of(Sirius.getSettings()
```
Comment (sabieber) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: Ich finde das als Inline Parameter richtig schrecklich. Würde vorschlagen das in ne Variable auszulagern

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 3]
```diff
@@ -115,7 +124,11 @@ public static SMTPConfiguration fromUserSettings() {
     public static SMTPConfiguration fromSettings(Settings settings) {
         return new SMTPConfiguration(settings.get("mail.host").getString(),
                                      settings.get("mail.port").getString(),
-                                     settings.get("mail.protocol").asEnum(SMTPProtocol.class),
+                                     settings.get("mail.protocol")
+                                             .getEnum(SMTPProtocol.class)
+                                             .orElseGet(() -> Value.of(settings.get("mail.protocol").toUpperCase())
```
Comment (sabieber) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: Siehe oben

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 4]
```diff
@@ -63,6 +63,8 @@
      * @param mailSenderName           the sender name
      * @param useSenderAndEnvelopeFrom whether to fill the "Sender" and the "Envelope-From" header
      */
+    @SuppressWarnings("squid:S00107")
+    @Explain("All settings are needed for a proper SMTP setting")
     public SMTPConfiguration(String host,
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: maybe refactor to a builder pattern using a chain of withXXX "setters"

Comment (andyHa) [reply src/main/java/sirius/web/mails/SMTPConfiguration.java]: this is a parameter object already anyway

File: src/main/java/sirius/web/mails/SMTPConfiguration.java [hunk 5]
```diff
@@ -87,6 +89,16 @@ public SMTPConfiguration(String host,
      * @return a new configuration based on the config files.
      */
     public static SMTPConfiguration fromConfig() {
+
+        SMTPProtocol smtpProtocol = Sirius.getSettings()
```
Comment (andyHa) [inline src/main/java/sirius/web/mails/SMTPConfiguration.java]: extract into method Value -> SMTPProtocol "asProtocol" or the like..

**Pull request merged**